artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -576,23 +576,30 @@ public void lateDeliveryRollback(PagePosition position) {
 
    @Override
    public boolean isComplete(long page) {
-      logger.tracef("%s isComplete %d", this, page);
+      if (empty && consumedPages.isEmpty()) {
+         logger.tracef("%s isComplete %d", this, page);
+      }
+
       synchronized (consumedPages) {
          if (empty && consumedPages.isEmpty()) {
             if (logger.isTraceEnabled()) {
-               logger.tracef("isComplete(%d)::Subscription %s has empty=%s, consumedPages.isEmpty=%s", (Object)page, this, consumedPages.isEmpty());
+               logger.tracef("isComplete(%d)::Subscription %s has empty=%s, consumedPages.isEmpty=%s", page, this, empty, consumedPages.isEmpty());
             }
             return true;
          }
 
          PageCursorInfo info = consumedPages.get(page);
 
          if (info == null && empty) {
-            logger.tracef("isComplete(%d)::::Couldn't find info and it is empty", page);
+            if (empty && consumedPages.isEmpty()) {
+               logger.tracef("isComplete(%d)::::Couldn't find info and it is empty", page);
+            }
             return true;
          }
          else {
-            logger.tracef("isComplete(%d)::calling is %s", (Object)page, this, consumedPages.isEmpty());
+            if (empty && consumedPages.isEmpty()) {
```

Comment:
I don't understand what you meant by being homogenous.. I don't see any other if (empty && consumedPages.isEmpty()) on any other loggers...


It seems that you meant if (logger.istTraceEnabled())?


which I have been replacing by simple calls to tracef alone since that wouldn't make any big conversions.. and it's fairly cheap.